integration-tests/utils/integration-helpers.js

```diff
@@ -133,22 +133,62 @@ const deleteWorkspace = withSignedInPage(async ({ page, billingProject, workspac
   }
 });
 
-const deleteWorkspaceV2 = withSignedInPage(async ({ page, billingProject, workspaceName }) => {
+/* TODO
+const deleteWorkspaceInUi = async ({ page, testUrl, token, workspaceName }) => {
+  await gotoPage(page, testUrl);
+  await findText(page, 'View Workspaces');
+  await viewWorkspaceDashboard(page, token, workspaceName);
+  await findText(page, clickable({ textContains: 'Delete' }));
+  await noSpinnersAfter(page, {
+    action: () => click(page, clickable({ textContains: 'Delete' })),
+    timeout: Millis.ofMinute,
+    debugMessage: 'deleteWorkspaceInUi - opened delete',
+  });
+  await findText(page, "Please type 'Delete Workspace' to continue:");
+  await fillIn(page, input({ placeholder: 'Delete Workspace' }), 'Delete Workspace');
+  await noSpinnersAfter(page, {
+    action: () => click(page, clickable({ textContains: 'Delete Workspace' })),
+    timeout: Millis.ofMinutes(2),
+    debugMessage: 'deleteWorkspaceInUi - submitted delete',
+  });
+};
+*/
+
+const deleteWorkspaceV2AsUser = async ({ page, billingProject, workspaceName }) => {
   try {
-    await page.evaluate(
-      async (name, billingProject) => {
-        await window.Ajax().Workspaces.workspaceV2(billingProject, name).delete();
-      },
-      workspaceName,
-      billingProject
-    );
-    console.info(`Deleted workspace: ${workspaceName}`);
+    const isAppsEmpty = await deleteAppsV2({ page, billingProject, workspaceName });
+    const isRuntimesEmpty = await deleteRuntimesV2({ page, billingProject, workspaceName });
+    if (isAppsEmpty && isRuntimesEmpty) {
+      // TODO [IA-4337] fix disk delete as part of runtime delete
```

Comment:
Will this handle the provisioning state?

I'm worried about how a 3-minute delay will impact the integration tests. Instead of doing the delay, could we just skip deleting this workspace and know that we will attempt to delete it again the next time the integration tests run?

Reply:
Oh wait, this is the condition of apps and runtimes ACTUALLY being empty. So does this mean we do a 3-minute delay every time we try to delete a workspace??

Reply:
Cripes, you're right. Extremely wrong. This was there to skirt a known bug, but that shouldn't be happening here anyway (even done correctly). Will remove